polygon/bridge/db.go

```diff
@@ -0,0 +1,148 @@
+package bridge
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"time"
+
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon/polygon/heimdall"
+)
+
+// GetLatestEventID the latest state sync event ID in given DB, 0 if DB is empty
+// NOTE: Polygon sync events start at index 1
+func GetLatestEventID(db kv.RoDB) (uint64, error) {
```

Comment:
all of the functions in this file should accept ctx as an input and pass to db.View and db.Update instead of context.Background